dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/config/DimensionDataCloudControlComputeServiceContextModule.java

```diff
@@ -0,0 +1,254 @@
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new NetworkDomainState(api.getNetworkApi(), State.DELETED), timeouts.imageAvailable,
+            pollPeriod.pollInitialPeriod, pollPeriod.pollMaxPeriod);
+   }
+
+   @Provides
+   @Named(NETWORK_DOMAIN_NORMAL_PREDICATE)
+   protected Predicate<String> provideNetworkDomainNormalPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new NetworkDomainState(api.getNetworkApi(), State.NORMAL), timeouts.imageAvailable,
+            pollPeriod.pollInitialPeriod, pollPeriod.pollMaxPeriod);
+   }
+
+   @Provides
+   @Named(SERVER_STARTED_PREDICATE)
+   protected Predicate<String> provideServerStartedPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new ServerStatus(api.getServerApi(), true, true), timeouts.imageAvailable,
+            pollPeriod.pollInitialPeriod, pollPeriod.pollMaxPeriod);
+   }
+
+   @Provides
+   @Named(SERVER_STOPPED_PREDICATE)
+   @VisibleForTesting
+   public Predicate<String> provideServerStoppedPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new ServerStatus(api.getServerApi(), false, true), timeouts.imageAvailable,
+            pollPeriod.pollInitialPeriod, pollPeriod.pollMaxPeriod);
+   }
+
+   @Provides
+   @Named(SERVER_DELETED_PREDICATE)
+   @VisibleForTesting
+   public Predicate<String> provideServerDeletedPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new ServerState(api.getServerApi(), State.DELETED), timeouts.imageAvailable,
+            pollPeriod.pollInitialPeriod, pollPeriod.pollMaxPeriod);
+   }
+
+   @Provides
+   @Named(SERVER_NORMAL_PREDICATE)
+   protected Predicate<String> provideServerNormalPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new ServerState(api.getServerApi(), State.NORMAL), timeouts.imageAvailable,
+            pollPeriod.pollInitialPeriod, pollPeriod.pollMaxPeriod);
+   }
+
+   @Provides
+   @Named(VM_TOOLS_RUNNING_PREDICATE)
+   protected Predicate<String> provideVMToolsRunningPredicate(final DimensionDataCloudControlApi api,
+         final ComputeServiceConstants.Timeouts timeouts, final ComputeServiceConstants.PollPeriod pollPeriod) {
+      return retry(new VMToolsRunningStatus(api.getServerApi()), timeouts.imageAvailable, pollPeriod.pollInitialPeriod,
+            pollPeriod.pollMaxPeriod);
+   }
```

Comment:
In general, use the existing timeouts if they make sense, or use a Dimensiondata specific timeout. The timeouts configured in the `ComputeServiceConstants.Timeouts` class are automatically used by jclouds when waiting for a node being created. For example, perhaps there is no need to use the "nodeRunning" timeout for a server creation, as that is the timeout jclouds uses to wait for the entire process since the user calls "createNode" until it finishes creating it, which involves several additional steps.
Configure timeout values at your own criteria and provide the corresponding properties, with reasonable defaults, to let users configure them as needed.